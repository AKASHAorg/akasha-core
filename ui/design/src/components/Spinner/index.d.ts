import * as React from 'react';
export interface SpinnerProps {
  /** the size (width, height) of the spinner. @default 50 */
  size?: number;
  /** speed (in seconds). ex: 1.2 or 2. @default 1.2 */
  speed?: number;
  style?: React.CSSProperties;
}
declare const Spinner: React.FC<SpinnerProps>;
export default Spinner;
