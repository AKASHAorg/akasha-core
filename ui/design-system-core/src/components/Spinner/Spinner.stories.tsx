import React from 'react';
import Spinner, { ISpinnerProps } from './index';

export default {
  title: 'Spinner/Spinner',
  component: Spinner,
};

const Template = (args: ISpinnerProps) => <Spinner {...args} />;

export const SpinnerSmall = Template.bind({});
SpinnerSmall.args = {
  size: 'sm',
};

export const SpinnerSmallWithColor = Template.bind({});
SpinnerSmallWithColor.args = {
  color: 'green',
};

export const SpinnerLarge = Template.bind({});
SpinnerLarge.args = {
  size: 'lg',
};

export const SpinnerXXLarge = Template.bind({});
SpinnerXXLarge.args = {
  size: 'xxl',
};
