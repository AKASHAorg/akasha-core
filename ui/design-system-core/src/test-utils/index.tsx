import { fireEvent, queries, render } from '@testing-library/react';
import React, { ReactElement } from 'react';

export const wrapper = ({ children }: { children: React.ReactNode }) => {
  return <>{children}</>;
};

export const delay = (ms = 100) => new Promise(res => setTimeout(res, ms));

export const customRender = (ui: ReactElement, options: Record<string, unknown>) =>
  render(ui, { wrapper, queries: { ...queries }, ...options });

export const resizeWindow = (width: number, height: number) => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  window.innerWidth = width;
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  window.innerHeight = height;
  fireEvent(window, new Event('resize'));
};

export const createFile = (name: string, type = 'image/png') => {
  return new File(['test-file-content'], name, { type });
};
