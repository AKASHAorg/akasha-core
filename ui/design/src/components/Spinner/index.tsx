import * as React from 'react';
import styled, { keyframes } from 'styled-components';

const Wrapper = styled.div<{ minHeight: number }>`
  width: 100%;
  display: flex;
  min-height: ${props => props.minHeight}px;
  justify-content: center;
  padding: 0.625em 0 0 0;
`;

const rotator = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(270deg);
  }
`;

const dash = keyframes`
  0% {
    stroke-dashoffset: 187;
  }
  50% {
    stroke-dashoffset: ${187 / 4};
    transform: rotate(135deg);
  }
  100% {
    stroke-dashoffset: 187;
    transform: rotate(450deg);
  }
`;

// const colors = keyframes`
//   0% {
//     stroke: #026337;
//   }
//   25% {
//     stroke: #c5dbcf;
//   }
//   50% {
//     stroke: #003a63;
//   }
//   75% {
//     stroke: #c75c35;
//   }
//   100% {
//     stroke: #c7985c;
//   }
// `;

const myCircle = ({ className }: { className?: string }) => (
  <circle
    className={className}
    fill="none"
    strokeWidth="6"
    strokeLinecap="round"
    cx="33"
    cy="33"
    r="30"
  />
);

const StyledCircle = styled(myCircle)`
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  transform-origin: center;
  stroke: ${props => props.theme.colors.border};
  animation: ${dash} 1.4s ease-in-out infinite;
`;

const StyledSpinner = styled.svg`
  animation: ${rotator} 1.4s linear infinite;
  width: ${props => props.width}px;
  height: ${props => props.width}px;
`;

export interface SpinnerProps {
  /** the size (width, height) of the spinner. @default 50 */
  size?: number;
  /** speed (in seconds). ex: 1.2 or 2. @default 1.2 */
  speed?: number;
  style?: React.CSSProperties;
}

const Spinner: React.FC<SpinnerProps> = props => {
  const { size = 50 } = props; // , speed = 1.2
  return (
    <Wrapper minHeight={size} style={props.style}>
      <StyledSpinner width={size} height={size} viewBox="0 0 66 66">
        <StyledCircle />
      </StyledSpinner>
    </Wrapper>
  );
};
export default Spinner;
