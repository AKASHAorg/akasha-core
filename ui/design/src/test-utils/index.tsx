import { fireEvent, queries, render } from '@testing-library/react';
import { Grommet, ThemeType } from 'grommet';
import React, { ReactElement } from 'react';
import lightTheme from '../styles/themes/light/light-theme';

export const wrapWithTheme = (children: any) => {
  return <Grommet theme={lightTheme as ThemeType}>{children}</Grommet>;
};

export const WithProviders = ({ children }: { children: any }) => {
  return <Grommet theme={lightTheme as ThemeType}>{children}</Grommet>;
};

export const delay = (ms = 100) => new Promise(res => setTimeout(res, ms));

export const customRender = (ui: ReactElement, options: any) =>
  render(ui, { wrapper: WithProviders, queries: { ...queries }, ...options });

export const resizeWindow = (width: number, height: number) => {
  // @ts-ignore
  window.innerWidth = width;
  // @ts-ignore
  window.innerHeight = height;
  fireEvent(window, new Event('resize'));
};

export const createFile = (name: string, type = 'image/png') => {
  return new File(['test-file-content'], name, { type });
};
