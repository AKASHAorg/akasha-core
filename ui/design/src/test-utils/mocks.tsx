export interface MockFileReader {
  result: string;
  onerror(): void;
  onload(): void;
  readAsDataURL(): void;
}

export interface WindowWithFileReader extends Window {
  FileReader: any;
}

export class MockFileReader {
  // tslint:disable-next-line: no-empty
  public onerror() {}
  // tslint:disable-next-line: no-empty
  public onload() {}
  public readAsDataURL() {
    this.result = 'test-file-content-result';
    this.onload();
  }
}
