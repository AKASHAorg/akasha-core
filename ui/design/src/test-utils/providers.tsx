import * as React from 'react';
import { DefaultTheme } from '../styles/themes/interfaces';
import { ThemeSelector } from '../styles/themes/utils/theme-selector';
import { createTheme } from '../styles/themes/utils/create-theme';
// const { ThemeSelector, createTheme } = DS;

const TestThemeProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) => {
  const themeData = createTheme();
  return (
    <React.Suspense fallback="loading">
      <div id="test-theme-selector">
        <ThemeSelector
          availableThemes={[{ name: 'test-light-theme', theme: themeData as DefaultTheme }]}
          settings={{ activeTheme: 'lightTheme' }}
        >
          {children}
        </ThemeSelector>
      </div>
    </React.Suspense>
  );
};

export { TestThemeProvider };
