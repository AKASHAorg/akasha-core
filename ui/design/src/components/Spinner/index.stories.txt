/* eslint-disable import/first */
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import Spinner from './index';

storiesOf('Spinner', module).add('default', () => <Spinner size={25} />);
