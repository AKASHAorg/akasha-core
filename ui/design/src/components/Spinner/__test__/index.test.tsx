import * as React from 'react';
import { create } from 'react-test-renderer';
import Spinner from '../';
import { wrapWithTheme } from '../../../test-utils';

describe('Spinner component', () => {
  it('should match the snapshot', () => {
    const spinner = create(wrapWithTheme(<Spinner />));
    expect(spinner.toJSON()).toMatchSnapshot('spinner-snapshot');
  });
});
