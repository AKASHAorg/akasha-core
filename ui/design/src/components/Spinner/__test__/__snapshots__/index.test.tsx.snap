// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should match the snapshot: spinner-snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 fPtJnA"
>
  <div
    className="sc-bdVaJa kRSGDk"
  >
    <svg
      className="sc-htpNat bUCXzc"
      height={50}
      viewBox="0 0 66 66"
      width={50}
    >
      <circle
        className="sc-bwzfXH gLcutx"
        cx="33"
        cy="33"
        fill="none"
        r="30"
        strokeLinecap="round"
        strokeWidth="6"
      />
    </svg>
  </div>
</div>
`;
